src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala

```diff
@@ -146,17 +156,44 @@ class EthService(blockchain: Blockchain, blockGenerator: BlockGenerator, appStat
 
   def getWork(req: GetWorkRequest): ServiceResponse[GetWorkResponse] = {
     import io.iohk.ethereum.mining.pow.PowCache._
-    val block = blockGenerator.generateBlockForMining()
-    Future.successful(Right(GetWorkResponse(
-      powHeaderHash = ByteString(kec256(BlockHeader.getEncodedWithoutNonce(block.header))),
-      dagSeed = seedForBlock(block.header.number),
-      target = ByteString((BigInt(2).pow(256) / block.header.difficulty).toByteArray)
-    )))
+
+    val blockNumber = appStateStorage.getBestBlockNumber() + 1
+
+    import scala.concurrent.ExecutionContext.Implicits.global
+    implicit val timeout = Timeout(5 seconds)
+    (pendingTransactionsManager ? PendingTransactionsManager.GetPendingTransactions).map{
+      case PendingTransactions(txList) =>
+        val fakeAddress = 42
+        //todo ask for ommers
+        val ommersList = Nil
+        val block = blockGenerator.generateBlockForMining(blockNumber, txList, ommersList, Address(fakeAddress))
+
+        block match {
+          case Right(b) =>
+            Right(GetWorkResponse(
+              powHeaderHash = ByteString(kec256(BlockHeader.getEncodedWithoutNonce(b.header))),
+              dagSeed = seedForBlock(b.header.number),
+              target = ByteString((BigInt(2).pow(256) / b.header.difficulty).toByteArray)
+            ))
+          case Left(err) =>
+            log.error(s"unable to prepare block because of $err")
+            Left(JsonRpcErrors.InternalError)
+        }
+      case a =>
```

Comment:
you can just do `(pendingTransactionsManager ? PendingTransactionsManager.GetPendingTransactions).mapTo[PendingTransactoins].map { pendingTransactinons => ... }`
you don't have to handle this error then (future will fail automatically)